tsfc/kernel_interface/firedrake_loopy.py

```diff
@@ -259,14 +259,26 @@ def __init__(self, integral_data_info, scalar_type,
         if integral_type in ['exterior_facet', 'exterior_facet_vert']:
             facet = gem.Variable('facet', (1,))
             self._entity_number = {None: gem.VariableIndex(gem.Indexed(facet, (0,)))}
+            facet_orientation = gem.Variable('facet_orientation', (1,), dtype=gem.uint_type)
+            self._entity_orientation = {None: gem.VariableIndex(gem.Indexed(facet_orientation, (0,)))}
         elif integral_type in ['interior_facet', 'interior_facet_vert']:
             facet = gem.Variable('facet', (2,))
             self._entity_number = {
                 '+': gem.VariableIndex(gem.Indexed(facet, (0,))),
                 '-': gem.VariableIndex(gem.Indexed(facet, (1,)))
             }
+            facet_orientation = gem.Variable('facet_orientation', (2,), dtype=gem.uint_type)
+            self._entity_orientation = {
+                '+': gem.VariableIndex(gem.Indexed(facet_orientation, (0,))),
+                '-': gem.VariableIndex(gem.Indexed(facet_orientation, (1,)))
+            }
         elif integral_type == 'interior_facet_horiz':
             self._entity_number = {'+': 1, '-': 0}
+            facet_orientation = gem.Variable('facet_orientation', (1,), dtype=gem.uint_type)  # base mesh entity orientation
+            self._entity_orientation = {
+                '+': gem.VariableIndex(gem.Indexed(facet_orientation, (0,))),
+                '-': gem.VariableIndex(gem.Indexed(facet_orientation, (0,)))
```

Comment:
are these really both zero?

Reply:
They indeed are. This is for `interior_facet_horiz`, so when we walk up the column, facet orientation is always the same as the base cell orientation.